File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    sorted_size=[]
    for size in sizes:
        size.sort(reverse=True)
        sorted_size.append(size)
    max_r,max_w=0,0
    for s in sorted_size:
        w,r=s
        if w>max_w:
            max_w=w
        if r>max_r:
            max_r=r
    return max_r*max_w

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>
#include <iostream>
#include <queue>
#include <algorithm>
using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    priority_queue<int> row;
    priority_queue<int> col;
    int n=sizes.size();
    for(int i=0;i<n;i++){
        sort(sizes[i].begin(),sizes[i].end());
        row.push(sizes[i][0]);
        col.push(sizes[i][1]);
    }
    return row.top()*col.top();
}